src/triage/component/postmodeling/contrast/model_evaluator.py

```diff
@@ -165,96 +165,73 @@ def crosstabs(self):
-        as the pred_matrix attribute of the class.
+        Load predictions matrices using the catwalk.storage.ProjectStorage.
+        This class allow the user to define a project path that can be either a
+        system local path or an s3 path and it will handle itself this
+        different infrastructures. 
 
-        For using s3 you have to define a bucket name and a path inside that
-        bucket to allow boto3 to retrieve the file and read it without having
-        to copy into memory (pandas will read it)
+        Once defined, we can pass this object to the MatrixStorageEngine that
+        will read this object and return a MatrixSotre object with a set of
+        handy methods to handle matrices
 
         Arguments:
-            - top_n: Only retrieve predicitions for the top_n observations
-            based in score
-            - Arguments inherited from _fetch_matrices: 
-                - path: relative path to the triage matrices folder or s3 path
+            path: project path to initiate the ProjectStorage object
         '''
         cache = self.__dict__.setdefault('_preds_matrix_cache', {})
         try:
-            return cache[path, top_n]
+            return cache[path]
         except KeyError:
             pass
 
-        matrix_path = path + self.pred_matrix_uuid + '.csv'
-        if 's3' in matrix_path:
-            fs = s3fs.S3FileSystem()
-            try:
-                with fs.open(matrix_path) as s3_file:
-                    mat = pd.read_csv(s3_file)
-            except FileNotFoundError:
-                print('No file in Bucket')
-
-        else:
-            mat = pd.read_csv(matrix_path)
-
-        if top_n is None: 
-            # Merge feature/prediction matrix with 
-            merged_df = mat.merge(self.predictions,
-                                  on='entity_id',
-                                  how='inner',
-                                  suffixes=('test', 'pred'))
+        storage = ProjectStorage(path)
+        matrix_storage = MatrixStorageEngine(storage).get_store(self.pred_matrix_uuid)
+        mat = matrix_storage.matrix
 
-        else:
-            # Filter  to the top_n entities
-            self.predictions['above_tresh'] = np.where(self.predictions['rank_abs'] <=
-                                                   top_n, 1, 0)
-
-            # Merge features with top_n predicted scores
-            merged_df = mat.merge(self.predictions,
-                                on='entity_id',
-                                how='inner',
-                                suffixes=('test','pred'))
+        # Merge with predictions table and return complete matrix
+        merged_df = mat.merge(self.predictions,
+                             on='entity_id',
+                             how='inner',
+                             suffixes=('test', 'pred'))
 
-        cache[path, top_n] = merged_df
+        cache[path] = merged_df
         self.preds_matrix = merged_df
```

Comment:
I'm pretty sure you don't want to overwrite the reference to the method with the cached result.

At this point you've cached it. You can just return it to the caller.